lib/Parse/Lexer.cpp

```diff
@@ -1244,6 +1244,9 @@ static const char *skipToEndOfInterpolatedExpression(const char *CurPtr,
                                                      DiagnosticEngine *Diags,
                                                      bool MultilineString) {
   llvm::SmallVector<char, 4> OpenDelimiters;
+  llvm::SmallVector<bool, 3> AllowNewline;
```

Comment:
My thought was: if size of `OpenDelimiters` is 4, size of `AllowNewline` is at most 3.
```
str = "a\(  "b\(  "c\(  something  )")")"
            "  (  "  (   -> OpenDelimiters : {'"', '(', '"', '('}
         f  f     f      -> AllowNewline : {false, false, false}
```
That was unnecessary restriction though. Updated to `4`.